contrib/android/include/android_lf.h

```diff
@@ -0,0 +1,39 @@
+/* 
+ * Macros for file64 functions
+ *
+ * Android does not support the macro _FILE_OFFSET_BITS=64
+ * As of android-21 it does however support many file64 functions
+*/
+
+/* EXPLICITLY NO HEADER GUARD */

```

Comment:
I don't know if this would work, but it would be simpler:

Put these at the top of *this* file:

#include <sys/types.h>
#include <unistd.h>
#include <fcntl.h>
... etc ...

That way, the relevant symbols will be defined before you override them.  Those system headers should have include guards, so this will ensure that they're included before your macros rearrange things.